From 915869e15ecfdb91e450f70825245c21032ee277 Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Tue, 6 Jun 2017 14:23:14 +1200
Subject: [PATCH] PUBAPI-1399: fabrics.test.js failing with multi-process
 testing due to caching Reviewed by: Trent Mick <trent.mick@joyent.com>

---
 lib/config.js | 12 ++++++++++--
 package.json  |  2 +-
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/lib/config.js b/lib/config.js
index 64b638a..f9ef9e2 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -139,7 +139,15 @@ function getConfigFromUFDS(req, callback) {
         req.log.debug({ account: account, dc: dc, config: conf },
             'Got user config');
         return callback(null, translateUfdsConf(conf));
-    });
+    }, true); // disable caching
+    /*
+     * Fetching the config is an uncommon operation, and caching the value
+     * causes inconsistencies since cloudapi usually operates with multiple
+     * processes (thus multiple caches, where invalidations and updates in one
+     * process don't occur in the others). Disabling caching ensures that
+     * clients get a consistent view, and prevents problems in other parts of
+     * cloudapi's code that uses values in the config.
+     */
 }
 
 
diff --git a/package.json b/package.json
index 85d25b1..1638e19 100644
--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
         "cueball": "2.1.1",
         "kang": "1.1.0",
         "sdc-clients": "10.2.0",
-        "ufds": "1.2.0",
+        "ufds": "1.3.0",
         "semver": "2.2.1",
         "nodemailer": "0.3.29",
         "clone": "0.1.5",
-- 
2.21.0

